integration/v3_election_test.go

```diff
@@ -225,3 +225,39 @@ func TestElectionOnPrefixOfExistingKey(t *testing.T) {
 		t.Fatal(err)
 	}
 }
+
+// TestElectionOnSessionRestart tests that a quick restart of leader (resulting
+// in a new session with the same lease id) does not result in loss of
+// leadership.
+func TestElectionOnSessionRestart(t *testing.T) {
+	clus := NewClusterV3(t, &ClusterConfig{Size: 1})
+	defer clus.Terminate(t)
+	cli := clus.RandClient()
+
+	session, err := concurrency.NewSession(cli)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	e := concurrency.NewElection(session, "test-elect")
+	if err1 := e.Campaign(context.TODO(), "abc"); err1 != nil {
+		t.Fatal(err1)
+	}
+
```

Comment:
can this also test that a waiter is never elected on the failover by having a waiter right here?

```go
// ensure leader is not lost to waiter on fail-over
waitSession, werr := concurrency.NewSession(cli)
if werr != nil {
    t.Fatal(werr)
}
defer waitSession.Orphan()
waitCtx, waitCancel := context.WithCancel(context.TODO())
defer waitCancel()
go concurrency.NewElection(waitSession, "test-elect").Campaign(waitCtx, "123")
```